### PR TITLE
Release v0.1.0

### PR DESCRIPTION
**Pull request by mehdichaouch**

### CODE REVIEW

File: app/code/community/Oyst/OneClick/Block/Adminhtml/Config/Mode.php [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+ */
+class Oyst_OneClick_Block_Adminhtml_Config_Mode extends Mage_Adminhtml_Block_System_Config_Form_Field
+{
+    protected function _getElementHtml(Varien_Data_Form_Element_Abstract $element)
+    {
+        $html = parent::_getElementHtml($element);
+
+        $javascript = "
+        <script type=\"text/javascript\">
+            element = $('" . $element->getHtmlId() . "');
```
Comment (jney) [inline app/code/community/Oyst/OneClick/Block/Adminhtml/Config/Mode.php]: if it is a local:
```js
var element = //...
```

if it is a global:
```js
window.element = //...
```

File: app/code/community/Oyst/OneClick/Block/Adminhtml/Config/Mode.php [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+            Event.observe(element, 'change', function(){
+                if (element.selectedOptions[0].value != '" . $element->getValue() . "'){
+                    $('oneclick_mode_comment').style.display = 'block';
+                } else {
+                    $('oneclick_mode_comment').style.display = 'none';
```
Comment (jney) [inline app/code/community/Oyst/OneClick/Block/Adminhtml/Config/Mode.php]:
```js
$('oneclick_mode_comment').hide()
```

File: app/code/community/Oyst/OneClick/Block/Adminhtml/Config/Mode.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    protected function _getElementHtml(Varien_Data_Form_Element_Abstract $element)
+    {
+        $html = parent::_getElementHtml($element);
+
+        $javascript = "
+        <script type=\"text/javascript\">
+            element = $('" . $element->getHtmlId() . "');
+            Event.observe(element, 'change', function(){
+                if (element.selectedOptions[0].value != '" . $element->getValue() . "'){
+                    $('oneclick_mode_comment').style.display = 'block';
```
Comment (jney) [inline app/code/community/Oyst/OneClick/Block/Adminhtml/Config/Mode.php]:
```js
 $('oneclick_mode_comment').show()
```

File: app/code/community/Oyst/OneClick/Block/Adminhtml/Config/Mode.php [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+                } else {
+                    $('oneclick_mode_comment').style.display = 'none';
+                }
+            });
+        </script>";
```
Comment (jney) [inline app/code/community/Oyst/OneClick/Block/Adminhtml/Config/Mode.php]: global comment about css selector namming, prefer `-` over `_` or camelCase

File: app/design/frontend/base/default/template/oyst/oneclick.phtml [hunk 5]
```diff
@@ -0,0 +1,71 @@
...
+if ($catalogHelper->isSupportedProduct($product)) :
+?>
+    <div id="oyst-1click-button"></div>
+    <script>
+        (function () {
+            window.__OYST__ = window.__OYST__ || {};
+            window.__OYST__.getOneClickURL = function (cb, opts) {
+                opts = opts || {};
+
+                ready(function () {
```
Comment (jney) [inline app/design/frontend/base/default/template/oyst/oneclick.phtml]: what is `ready` function? I mean, where is it declared?

File: app/design/frontend/base/default/template/oyst/oneclick.phtml [hunk 6]
```diff
@@ -0,0 +1,71 @@
...
+                    var qty = $$("input[name='qty']")[0].value ? $$("input[name='qty']")[0].value : 1;
+
+                    var form = new FormData();
+                    form.append("productRef", <?php echo $product->getId(); ?>);
+                    form.append("variationRef", getSimpleProductId());
+                    form.append("quantity", qty);
+
+                    var settings = {
+                        "method": "POST",
```
Comment (jney) [inline app/design/frontend/base/default/template/oyst/oneclick.phtml]: useless quotes for the keys

File: app/design/frontend/base/default/template/oyst/oneclick.phtml [hunk 7]
```diff
@@ -0,0 +1,71 @@
...
+                        "url": "<?php echo $oneClickUrl ?>",
+                        "async": true,
+                        'cacheControl': "no-cache",
+                        "mimeType": "multipart/form-data",
+                        "data": form
```
Comment (jney) [inline app/design/frontend/base/default/template/oyst/oneclick.phtml]: sort keys

File: js/oyst/oneclick.js [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+/**
+ * OneClick Customs JS
+ */
+
+function showMessage(txt, type) {
+    var html = '<ul class="messages"><li class="' + type + '-msg"><ul><li>' + txt + '</li></ul></li></ul>';
+    $("messages").update(html);
+    var url = location.href;
+    var n = url.indexOf("#");
+    url = url.substring(0, n != -1 ? n : url.length);
```
Comment (jney) [inline js/oyst/oneclick.js]: those 3 lines can be simplified to

```js
var url = location.href.split('#')[0]
```

File: app/code/community/Oyst/OneClick/Block/Adminhtml/System/Config/Form/Field/Config.php [hunk 9]
```diff
@@ -0,0 +1,91 @@
...
+
+/**
+ * Adminhtml System Config Form Field Config Block
+ */
+class Oyst_OneClick_Block_Adminhtml_System_Config_Form_Field_Config extends Mage_Adminhtml_Block_System_Config_Form_Field
+{
+    protected function _getElementHtml(Varien_Data_Form_Element_Abstract $element)
+    {
+        $trad = array(
+            'postShimpentsSuccess' => $this->__('Delivery mode synchronization was successful.'),
```
Comment (nicolaschenet) [inline app/code/community/Oyst/OneClick/Block/Adminhtml/System/Config/Form/Field/Config.php]: Do you mean `postShipmentsSuccess` ?

File: app/code/community/Oyst/OneClick/Block/Adminhtml/System/Config/Form/Field/Config.php [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+                #oyst_oneclick_oneclick_shipments_config { font-family: courier, courier new, serif; height: 30em; }
+            </style>
+            <script type="text/javascript">
+            //<![CDATA[
+                function showMessage(txt, type) {
+                    var html = \'<ul class="messages"><li class="\' + type + \'-msg"><ul><li>\' + txt + \'</li></ul></li></ul>\';
+                    $("messages").update(html);
+                    var url = location.href;
+                    var n = url.indexOf("#");
+                    url = url.substring(0, n != -1 ? n : url.length);
```
Comment (nicolaschenet) [inline app/code/community/Oyst/OneClick/Block/Adminhtml/System/Config/Form/Field/Config.php]: `!==`

File: app/code/community/Oyst/OneClick/Block/Adminhtml/System/Config/Form/Field/Config.php [hunk 11]
```diff
@@ -0,0 +1,91 @@
...
+                        Element.hide("loading-mask");
+                        return;
+                    }
+
+                    var url = "' . $this->getUrl('adminhtml/oneclick_catalog/postShipments') . '";
+                    var myAjax = new Ajax.Request(url, {
+                        method: "post",
+                        onComplete: function (xhr) {
+                            Element.hide("loading-mask");
+                            if (200 == xhr.status && "" == xhr.responseText) {
```
Comment (nicolaschenet) [inline app/code/community/Oyst/OneClick/Block/Adminhtml/System/Config/Form/Field/Config.php]: `===`

File: app/design/frontend/base/default/template/oyst/oneclick.phtml (hunk 5 above)
Comment (nicolaschenet) [reply app/design/frontend/base/default/template/oyst/oneclick.phtml]: the same, where does it come from ?

File: skin/frontend/base/default/js/oyst/oneclick.js [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+    }
+}
+
+/**
+ * Return the simple product id from a configurable product
+ *
+ * @returns {null}
+ */
+function getSimpleProductId() {
+    if (typeof(spConfig) == 'undefined') {
```
Comment (nicolaschenet) [inline skin/frontend/base/default/js/oyst/oneclick.js]: generally speaking, prefer `===`

File: skin/frontend/base/default/js/oyst/oneclick.js [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+        return null;
+    }
+
+    var productCandidates = [];
+
+    spConfig.settings.forEach(function (select, selectIndex) {
+        var attributeId = select.id.replace('attribute', '');
+        var selectedValue = select.options[select.selectedIndex].value;
+
+        spConfig.config.attributes[attributeId].options.forEach(function (option, optionIndex) {
```
Comment (nicolaschenet) [inline skin/frontend/base/default/js/oyst/oneclick.js]: you could consider using `Array.prototype.filter`
https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/Array/filter

File: skin/frontend/base/default/js/oyst/oneclick.js [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+            if (option.id == selectedValue) {
+                var optionProducts = option.products;
+
+                if (0 == productCandidates.length) {
```
Comment (nicolaschenet) [inline skin/frontend/base/default/js/oyst/oneclick.js]: When can  `productCandidates.length` be different from 0 ?

File: app/code/community/Oyst/OneClick/Block/Adminhtml/Config/Mode.php (hunk 2 above)
Comment (mehdichaouch) [reply app/code/community/Oyst/OneClick/Block/Adminhtml/Config/Mode.php]: Fixed.

File: app/code/community/Oyst/OneClick/Block/Adminhtml/Config/Mode.php (hunk 1 above)
Comment (mehdichaouch) [reply app/code/community/Oyst/OneClick/Block/Adminhtml/Config/Mode.php]: Fixed.

File: app/code/community/Oyst/OneClick/Block/Adminhtml/Config/Mode.php (hunk 3 above)
Comment (mehdichaouch) [reply app/code/community/Oyst/OneClick/Block/Adminhtml/Config/Mode.php]: Fixed.

File: app/code/community/Oyst/OneClick/Block/Adminhtml/Config/Mode.php (hunk 4 above)
Comment (mehdichaouch) [reply app/code/community/Oyst/OneClick/Block/Adminhtml/Config/Mode.php]: Ok.

File: app/design/frontend/base/default/template/oyst/oneclick.phtml (hunk 5 above)
Comment (mehdichaouch) [reply app/design/frontend/base/default/template/oyst/oneclick.phtml]: from `skin/frontend/base/default/js/oyst`: [here](https://github.com/oystparis/oyst-1click-magento/blob/v0.1.0/skin/frontend/base/default/js/oyst/oneclick.js)

File: app/design/frontend/base/default/template/oyst/oneclick.phtml (hunk 6 above)
Comment (mehdichaouch) [reply app/design/frontend/base/default/template/oyst/oneclick.phtml]: Fixed.

File: app/design/frontend/base/default/template/oyst/oneclick.phtml (hunk 7 above)
Comment (mehdichaouch) [reply app/design/frontend/base/default/template/oyst/oneclick.phtml]: Fixed.

File: js/oyst/oneclick.js (hunk 8 above)
Comment (mehdichaouch) [reply js/oyst/oneclick.js]: Fixed.

File: app/code/community/Oyst/OneClick/Block/Adminhtml/System/Config/Form/Field/Config.php (hunk 9 above)
Comment (mehdichaouch) [reply app/code/community/Oyst/OneClick/Block/Adminhtml/System/Config/Form/Field/Config.php]: Yes of course ;)
Fixed.

File: app/code/community/Oyst/OneClick/Block/Adminhtml/System/Config/Form/Field/Config.php (hunk 11 above)
Comment (mehdichaouch) [reply app/code/community/Oyst/OneClick/Block/Adminhtml/System/Config/Form/Field/Config.php]: Fixed.

File: app/code/community/Oyst/OneClick/Block/Adminhtml/System/Config/Form/Field/Config.php (hunk 10 above)
Comment (mehdichaouch) [reply app/code/community/Oyst/OneClick/Block/Adminhtml/System/Config/Form/Field/Config.php]: I changed theses three lines for @jney suggestion.

File: skin/frontend/base/default/js/oyst/oneclick.js (hunk 12 above)
Comment (mehdichaouch) [reply skin/frontend/base/default/js/oyst/oneclick.js]: Fixed.

File: skin/frontend/base/default/js/oyst/oneclick.js (hunk 13 above)
Comment (mehdichaouch) [reply skin/frontend/base/default/js/oyst/oneclick.js]: Ok.

File: skin/frontend/base/default/js/oyst/oneclick.js (hunk 14 above)
Comment (mehdichaouch) [reply skin/frontend/base/default/js/oyst/oneclick.js]: After the first loop.

**Pull request merged**